ehrql/utils/math_utils.py

```diff
@@ -16,3 +16,10 @@ def floordiv(lhs, rhs):
         return None
     else:
         return int(lhs // rhs)
+
+
+def count_distinct(series):
+    """
+    Count distinct values in a series.
+    """
+    return len(set(series))
```

Comment:
You asked whether this was the right place for this function and I'm not sure it is. I think it would actually be simpler and clearer to define it inline in `InMemoryQueryEngine`. There are already a few examples of this e.g.:
https://github.com/opensafely-core/ehrql/blob/ad6ac4a389d0be2a2a0faf918e4736aa324df2ed/ehrql/query_engines/in_memory.py#L292-L296

In fact, I think we should probably move the other math_utils functions inline as well, but that's a job for a different PR.

The reason we initially had them separate like this was that they got used in both the in-memory engine and in the population validation module. But I've since simplified the population validation code to re-use the logic from the in-memory engine, so we no longer need to define behaviour in two places.